src/main/java/com/icthh/xm/gate/gateway/ratelimiting/RateLimitingConfiguration.java

```diff
@@ -36,22 +35,26 @@ public KeyResolver tenantKeyResolver() {
 
     @Bean
     public KeyResolver tenantClientKeyResolver() {
-        return exchange -> Mono.just(getClientIdFromToken(exchange.getRequest()));
+        return exchange -> Mono.just(getClientIdKey(exchange.getRequest()));
     }
 
-    private String getClientIdFromToken(ServerHttpRequest request) {
+    @Bean
+    public KeyResolver serviceSessionIdKeyResolver() {
+        return exchange -> Mono.just(getSessionIdKey(exchange.getRequest()));
+    }
+
+    private String getSessionIdKey(ServerHttpRequest request) {
```

Comment:
if request.header contains X-SESSIONID then take value, otherwise ""
if value = "" filter should not work, otherwise evaluate "service:SESSION-ID" limit